eng/tools/openapi-diff-runner/test/utils/spec.test.ts

```diff
@@ -266,4 +267,96 @@ describe("Helper functions for version analysis", () => {
       expect(result).toBe("test.json");
     });
   });
+
+  describe("deduplicateSwaggers", () => {
+    it("should return empty array for empty input", () => {
+      const result = deduplicateSwaggers([]);
+      expect(result).toEqual([]);
+    });
+
+    it("should return the same array when no duplicates exist", () => {
+      const mockSwaggers: MockSwagger[] = [
+        createMockSwagger(TEST_PATHS.stable2020_07_02),
+        createMockSwagger(TEST_PATHS.stable2020_08_04),
+        createMockSwagger(TEST_PATHS.preview2020_07_02),
+      ];
+
+      const result = deduplicateSwaggers(mockSwaggers);
+
+      expect(result).toHaveLength(3);
+      expect(result[0].path).toBe(TEST_PATHS.stable2020_07_02);
+      expect(result[1].path).toBe(TEST_PATHS.stable2020_08_04);
+      expect(result[2].path).toBe(TEST_PATHS.preview2020_07_02);
+    });
+
+    it("should remove duplicate swaggers with same path", () => {
+      const mockSwaggers: MockSwagger[] = [
+        createMockSwagger(TEST_PATHS.stable2020_07_02),
+        createMockSwagger(TEST_PATHS.stable2020_08_04),
+        createMockSwagger(TEST_PATHS.stable2020_07_02), // duplicate
+        createMockSwagger(TEST_PATHS.preview2020_07_02),
+        createMockSwagger(TEST_PATHS.stable2020_08_04), // duplicate
+      ];
+
+      const result = deduplicateSwaggers(mockSwaggers);
+
+      expect(result).toHaveLength(3);
+      expect(result.map((s) => s.path)).toEqual([
+        TEST_PATHS.stable2020_07_02,
+        TEST_PATHS.stable2020_08_04,
+        TEST_PATHS.preview2020_07_02,
+      ]);
+    });
+
+    it("should preserve the first occurrence when duplicates exist", () => {
```

Comment:
Do you actually need this behavior?  If so, I think we need to sort the inputs, since there are no guarantees about ordering by default